crates/bevy_core/src/time/timer.rs

```diff
@@ -4,39 +4,56 @@ use bevy_property::Properties;
 use std::time::Duration;
 
 /// Tracks elapsed time. Enters the finished state once `duration` is reached.
+///
+/// Non repeating timers will stop tracking and stay in the finished state until reset.
+/// Repeating timers will only be in the finished state on each tick `duration` is reached or exceeded, and can still be reset at any given point.
 #[derive(Clone, Debug, Default, Properties)]
 pub struct Timer {
     pub elapsed: f32,
     pub duration: f32,
     pub finished: bool,
+    prev_finished: bool,
+    pub repeating: bool,
 }
 
 impl Timer {
-    pub fn new(duration: Duration) -> Self {
+    pub fn new(duration: Duration, repeating: bool) -> Self {
         Timer {
             duration: duration.as_secs_f32(),
+            repeating,
             ..Default::default()
         }
     }
 
-    pub fn from_seconds(seconds: f32) -> Self {
+    pub fn from_seconds(seconds: f32, repeating: bool) -> Self {
         Timer {
             duration: seconds,
+            repeating,
             ..Default::default()
         }
     }
 
     pub fn tick(&mut self, delta: f32) {
-        self.elapsed = (self.elapsed + delta).min(self.duration);
-        if self.elapsed >= self.duration {
-            self.finished = true;
+        if self.elapsed < self.duration {
```

Comment:
By removing this _if_ check, elapsed could be used to calculate how long ago did the timer finish + a minor perf boost.
Or is there a concern for the float to overflow?

Reply:
I think it's more about expected behavior; if I set a timer for 10 seconds I expect it to notify me after 10 seconds and stop doing anything after that, but maybe that's just me.

I did have overflow in mind adding the check but I didn't check how long you'd actually have to leave it running. Since it's going to run out at some point anyways you're probably better off storing an `Instant` externally and comparing that later on.

Reply:
Wouldn't it cause just_finished to stay true even after ticks of it finishing? (as a non repeating timer)